### PR TITLE
BI-1306-[Buildbot]-Create-rspec-test-for-queue-function

### PR DESCRIPTION
**Pull request by hiepky**
Implement 4 unittest testcases:

- replies all found running builds if wildcard = False
- replies the correct running builds if wildcard = True
- replies "runningbuildtypes.empty" when no have queue was found if call "Queue"
- replies "error.request" if connection to Teamcity is error

### CODE REVIEW

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 1]
```diff
@@ -195,6 +195,78 @@
       expect(output_string).to eq(expected_output)
     end
   end
+  
+  describe '#format_time' do
```
Comment (nguyentranlogigear) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb:199]: This method does not mention in ticket. So could you help me show reason or remove it from PR

Comment (hiepky) [reply workspace/spec/lita/handlers/teamcity_spec_anki.rb:199]: I resloved conflict and edit this. this is not changed

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 2]
```diff
@@ -195,6 +195,78 @@
...
+                              "build": [{"buildTypeId": "CozmoOne_MasterBuild",
+                                        "branchName": "BI-123-XYZ",
+                                        "id": "4307"},
+                                       {"buildTypeId": "OverDriveOne_Release_Build",
+                                        "branchName": "BI-124-XYZ",
+                                        "id": "4308"}]}'
+    before do
+     stub_request(:get, /buildQueue/).to_return(:status => 200, :body => queue_build_json_str)
+    end
+    it 'Reply all queue builds are found if call "Queue"' do
```
Comment (nguyentranlogigear) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: I think that we should have break line between end and it

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 3]
```diff
@@ -195,6 +195,78 @@
...
+
+    end
+
+    it 'Reply empty if call "Queue_Wildcard" with invalid wildcard' do
+      expected_output = nil
+      send_command('queue abc')
+      expect(replies.last).to eq(expected_output)
+
+    end
+    it 'The message display when no have queue is found if call "Queue"' do
```
Comment (nguyentranlogigear) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: I think that we should have break line between end and it

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 4]
```diff
@@ -195,6 +195,78 @@
...
+    end
+    it 'Reply all queue builds are found if call "Queue"' do
+      expected_output = "Here are the builds I found:\n"\
+                        "\n<#{registry.config.handlers.teamcity.site}/viewQueued.html?itemId=4307"\
+                        "|CozmoOne_MasterBuild> - BI-123-XYZ"\
+                        "\n<#{registry.config.handlers.teamcity.site}/viewQueued.html?itemId=4308"\
+                        "|OverDriveOne_Release_Build> - BI-124-XYZ"
+      send_command('queue')
+      expect(replies.last).to eq(expected_output)
+
```
Comment (nguyentranlogigear) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: I think that we should remove blank line before end line

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 5]
```diff
@@ -195,6 +195,76 @@
...
+      expect(replies.last).to eq(expected_output)
+    end
+
+    it 'Reply empty if call "Queue_Wildcard" with invalid wildcard' do
+      expected_output = nil
+      send_command('queue abc')
+      expect(replies.last).to eq(expected_output)
+    end
+    
+    it 'The message display when no have queue is found if call "Queue"' do
```
Comment (vovuong) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: Both the "it 'The message display when no have queue is found if call "Queue"' do" and "it 'Reply empty if call "Queue_Wildcard" with invalid wildcard' do" cases are returned an empty result. So only one case should be tested here

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 6]
```diff
@@ -195,6 +195,76 @@
...
+      send_command('queue')
+      expect(replies.last).to eq(expected_output)
+    end
+
+    it 'Reply all queue builds are found if call "Queue_Wildcard"' do
+      expected_output = "Here are the builds I found:\n"\
+                        "\n<#{registry.config.handlers.teamcity.site}/viewQueued.html?itemId=4307"\
+                        "|CozmoOne_MasterBuild> - BI-123-XYZ"
+
+      send_command('queue CozmoOne*')
```
Comment (vovuong) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: Should create an variable for the 'CozmoOne'.

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 7]
```diff
@@ -195,6 +195,76 @@
...
+                                        "branchName": "BI-123-XYZ",
+                                        "id": "4307"},
+                                       {"buildTypeId": "OverDriveOne_Release_Build",
+                                        "branchName": "BI-124-XYZ",
+                                        "id": "4308"}]}'
+    before do
+     stub_request(:get, /buildQueue/).to_return(:status => 200, :body => queue_build_json_str)
+    end
+    
+    it 'Reply all queue builds are found if call "Queue"' do
```
Comment (vovuong) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: The name of unittest test case should be more detailed.
Please refer https://github.com/scopp/lita-slack-teamcity/pull/29/files#diff-2246990822dced4151fb04ca044faed1R231

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 8]
```diff
@@ -195,6 +195,76 @@
...
+      expect(output_string).to eq("%Mm:%Ss")
+    end
+
+    it 'format_time with time <= 60' do
+      output_string = subject.format_time(60)
+      expect(output_string).to eq("%Ss")
+    end
+  end
+
+  describe '#queue' do
```
Comment (vovuong) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb:217]: Please add case "no connection to Teamcity"

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 9]
```diff
@@ -195,6 +195,76 @@
...
+      send_command('queue abc')
+      expect(replies.last).to eq(expected_output)
+    end
+    
+    it 'The message display when no have queue is found if call "Queue"' do
+      queue_build_json_str = '{ "count": 0,
+                              "build": [{"buildTypeId": "CozmoOne_MasterBuild",
+                                        "branchName": "BI-123-XYZ",
+                                        "id": "4307"},
+                                       {"buildTypeId": "OverDriveOne_Release_Build",
```
Comment (scopp) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: @vovuong delete all company specific information!!

**Review by scopp — CHANGES_REQUESTED** [review]

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb [hunk 10]
```diff
@@ -195,6 +195,76 @@
...
+
+    it 'format_time with time <= 60' do
+      output_string = subject.format_time(60)
+      expect(output_string).to eq("%Ss")
+    end
+  end
+
+  describe '#queue' do
+    queue_build_json_str = '{ "count": 2,
+                              "build": [{"buildTypeId": "CozmoOne_MasterBuild",
```
Comment (scopp) [inline workspace/spec/lita/handlers/teamcity_spec_anki.rb]: remove company specific info

**Review by scopp — CHANGES_REQUESTED** [review]
remove all company specific info

File: workspace/spec/lita/handlers/teamcity_spec_anki.rb (hunk 8 above)
Comment (nguyentranlogigear) [reply workspace/spec/lita/handlers/teamcity_spec_anki.rb:217]: I will add this case end of describe

**Comment by nguyentranlogigear** [comment]
LGTM

**Pull request merged**